modules/social_features/social_event/src/Plugin/GraphQL/SchemaExtension/EventSchemaExtension.php

```diff
@@ -0,0 +1,160 @@
+   *   The resolver registry.
+   * @param \Drupal\graphql\GraphQL\ResolverBuilder $builder
+   *   The resolver builder.
+   */
+  protected function addEventFields(ResolverRegistryInterface $registry, ResolverBuilder $builder) {
+    $registry->addFieldResolver('Event', 'id',
+      $builder->produce('entity_uuid')
+        ->map('entity', $builder->fromParent())
+    );
+
+    $registry->addFieldResolver('Event', 'title',
+      $builder->compose(
+        $builder->produce('entity_label')
+          ->map('entity', $builder->fromParent()),
+        $builder->produce('uppercase')
+          ->map('string', $builder->fromParent())
+      )
+    );
+
+    $registry->addFieldResolver('Event', 'author',
+      $builder->produce('entity_owner')
+        ->map('entity', $builder->fromParent())
+    );
+
+    $registry->addFieldResolver('Event', 'bodyHtml',
+      $builder->fromPath('entity:node', 'body.0.value')
+    );
+
+    $registry->addFieldResolver('Event', 'heroImage',
+      $builder->produce('field')
+        ->map('entity', $builder->fromParent())
+        ->map('field', $builder->fromValue('field_event_image'))
+    );
+
+    $registry->addFieldResolver('Event', 'created',
+      $builder->fromPath('entity:node', 'created.value')
+    );
+
+    $registry->addFieldResolver('Event', 'comments',
+      $builder->produce('social_comments')
+        ->map('parent', $builder->fromParent())
+        ->map('after', $builder->fromArgument('after'))
+        ->map('before', $builder->fromArgument('before'))
+        ->map('first', $builder->fromArgument('first'))
+        ->map('last', $builder->fromArgument('last'))
+        ->map('reverse', $builder->fromArgument('reverse'))
+        ->map('sortKey', $builder->fromArgument('sortKey'))
+    );
+
+    $registry->addFieldResolver('Event', 'startDate',
+      $builder->compose(
+        $builder->produce('field')
+          ->map('entity', $builder->fromParent())
+          ->map('field', $builder->fromValue('field_event_date')),
+        $builder->produce('date_to_timestamp')
+          ->map('field', $builder->fromParent())
+      )
+    );
+
+    $registry->addFieldResolver('Event', 'endDate',
+      $builder->compose(
+        $builder->produce('field')
+          ->map('entity', $builder->fromParent())
+          ->map('field', $builder->fromValue('field_event_date_end')),
+        $builder->produce('date_to_timestamp')
+          ->map('field', $builder->fromParent())
+          ->map('type', $builder->fromValue('end_date'))
+      )
+    );
+
+    $registry->addFieldResolver('Event', 'location',
+      $builder->fromPath('entity:node', 'field_event_location.value')
+    );
+
+    $registry->addFieldResolver('Event', 'url',
+      $builder->compose(
+        $builder->produce('entity_url')
+          ->map('entity', $builder->fromParent()),
+        $builder->produce('url_path')
+          ->map('url', $builder->fromParent())
```

Comment:
I think this field should produce an absolute URL correct? But this creates a relative path?

Reply:
Fixed.